src/lib/remote.sh

```diff
@@ -43,16 +43,21 @@ function ssh_connection_failure_message()
   local user=${remote_parameters['REMOTE_USER']}
   local remote_file=${remote_parameters['REMOTE_FILE']}
   local remote_file_host=${remote_parameters['REMOTE_FILE_HOST']}
-  local host_line_number
+  local target_host_line_number
+  local trimmed_remote_file
 
   # If a config file is being used, we need to set the IP, port
   # and user from the config file.
   if [[ -z "$remote" && -z "$port" && -z "$user" ]]; then
     if [[ -n "${remote_parameters['REMOTE_FILE']}" ]]; then
-      host_line_number=$(grep -n "Host ${remote_file_host}" "${remote_file}" | cut -d: -f1)
-      remote=$(sed -n "$((host_line_number + 1))p" "${remote_file}" | cut -d\  -f4)
-      port=$(sed -n "$((host_line_number + 2))p" "${remote_file}" | cut -d\  -f4)
-      user=$(sed -n "$((host_line_number + 3))p" "${remote_file}" | cut -d\  -f4)
+      target_host_line_number=$(grep -n "Host ${remote_file_host}" "${remote_file}" | cut -d: -f1)
+      trimmed_remote_file=$(tail --lines "+${target_host_line_number}" "${remote_file}")
+      # Here, we are matching the first line with Hostname/Port/User occurrence that happens after
+      # the `target_host_line_number` line and only have leading spaces (excludes commented lines).
+      # We then cut the resulting string to get only the value of the Hostname/Port/User line.
+      remote=$(printf '%s' "$trimmed_remote_file" | grep --max-count=1 '^[[:blank:]]*Hostname' | cut -d\  -f4)
```

Comment:
Come to think of it, would this regex (and the subsequent ones too) work if tabs were used, instead of white spaces, to indent the remote config file?

Reply:
Yes, I'm pretty sure that `[[:blank:]]` captures any whitespace (at least in my tests it worked 😆). Modified the sample file `remote_4.config` (used in the added unit test) to have lines with tabs, just to be sure. Thanks for the heads up!

Reply:
Interesting. It seems like the sample file only contains whitespaces still. I added a tab and the test fails. Could you double check that please?
![image](https://github.com/kworkflow/kworkflow/assets/48871985/46f19a77-6ee3-4b06-b651-298fed77004e)

Reply:
Hmm, interesting, this sample file does contains tabs, but the test does fails if we add more tabs or no tabs (or any number of whitespace that isn't 2). The problem actually is the `cut -d\  -f4` part of the command, as it assumes that there are exactly 2 leading whitespaces in the line. Going to fix that, thanks for digging deep in this matter 👍

Reply:
Hi @magalilemes fixed it. If possible, can you double check this for me?

Reply:
@davidbtadokoro Just tested and it works now.
However, what I've been meaning to say with tab is the tab character (ascii = 9). For instance, using vscode, if I select the lines I can see the difference between "real" tabs and whitespaces. I added a tab right before "Port 123" for demonstrating:
![image](https://github.com/kworkflow/kworkflow/assets/48871985/6bb685f0-9920-47ff-b6a4-6a1a4f307a5c)